core/src/test/kotlin/fr/sncf/osrd/standalone_sim/StandaloneSimulationTest.kt

```diff
@@ -286,6 +285,77 @@ class StandaloneSimulationTest {
         }
     }
 
+    /** Test that the safety speeds are properly computed for a given path */
+    @Test
+    fun testSafetySpeed() {
+        // Path length = 10400m
+        // SIGNAL OFFSETS on this path: 150m and 10150m
+        // Then, buffer stop at the end of the last route: 10400m
+        val schedule =
+            listOf(
+                // Start of path, check that it doesn't extend beyond the start
+                SimulationScheduleItem(
+                    Offset(42.meters),
+                    42.seconds,
+                    42.seconds,
+                    SHORT_SLIP_STOP,
+                ),
+                // OPEN, check that this is ignored
+                SimulationScheduleItem(
+                    Offset(10_000.meters),
+                    42.seconds,
+                    42.seconds,
+                    OPEN,
+                ),
+                // STOP before the last buffer stop
+                SimulationScheduleItem(
+                    Offset(10_300.meters),
+                    42.seconds,
+                    42.seconds,
+                    STOP,
+                ),
+            )
+        val safetySpeedRanges =
+            makeSafetySpeedRanges(infra, chunkPath, routes.toIdxList(), schedule)
+        val expected =
+            distanceRangeMapOf(
+                listOf(
+                    DistanceRangeMap.RangeMapEntry(0.meters, 50.meters, 30.kilometersPerHour),
+                    DistanceRangeMap.RangeMapEntry(50.meters, 150.meters, 10.kilometersPerHour),
+                    DistanceRangeMap.RangeMapEntry(
+                        10_200.meters,
+                        10_400.meters,
+                        30.kilometersPerHour
+                    ),
```

Comment:
If buffer-stops are automatically short-slip stops, this test should move (maybe also try to change the reception signal to `OPEN`)
```suggestion
                    DistanceRangeMap.RangeMapEntry(
                        10_200.meters,
                        10_300.meters,
                        30.kilometersPerHour
                    ),
                    DistanceRangeMap.RangeMapEntry(
                        10_300.meters,
                        10_400.meters,
                        10.kilometersPerHour
                    ),
```

Reply:
Will be part of https://github.com/OpenRailAssociation/osrd/issues/9283